ui/admin/app/routes/scopes/scope/roles/index.js

```diff
@@ -4,9 +4,87 @@
  */
 
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
 export default class ScopesScopeRolesIndexRoute extends Route {
+  // =attributes
+
+  queryParams = {
+    search: {
+      refreshModel: true,
+      replace: true,
+    },
+    page: {
+      refreshModel: true,
+    },
+    pageSize: {
+      refreshModel: true,
+    },
+  };
+
+  rolesExist;
+
+  // =services
+
+  @service can;
+  @service store;
+
   // =methods
 
+  /**
+   * Loads queried roles and the number of roles under current scope.
+   * @returns {Promise<{totalItems: number, users: [RoleModel], rolesExist: boolean }> }
+   */
+  async model({ search, page, pageSize }) {
+    const scope = this.modelFor('scopes.scope');
+    const { id: scope_id } = scope;
+    const filters = {
+      scope_id: [{ equals: scope_id }],
+    };
+
+    let roles;
+    let totalItems = 0;
+    if (this.can.can('list model', scope, { collection: 'roles' })) {
+      roles = await this.store.query('role', {
+        query: { search, filters },
+        page,
+        pageSize,
+      });
+      totalItems = roles.meta?.totalItems;
+    }
+    await this.getRolesExist(scope_id, totalItems);
```

Comment:
Why do we save this to a class variable `rolesExist` and then return it? Why couldn't you have just done `const rolesExist = this.getRolexExist` and have the result returned from the function?

Reply:
ah yes good point, ill refactor to do that instead!

Reply:
In reference to the manage dropdown: good catch, ur right that happens for roles + other resources. I added this [backlog ticket](https://hashicorp.atlassian.net/browse/ICU-13185) to fix this.